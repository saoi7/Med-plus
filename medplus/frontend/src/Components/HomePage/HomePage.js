import React, { useState } from 'react';
import { Link, Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { NavBar } from '../NavBar/NavBar';

// TODO this should redirect to LandingPage if user is not logged in

// TODO this needs to fetch medication list from server, display 'loading...' prompt while waiting for server response

// TODO this needs to update server when a medication is clicked (isTaken is toggled)

// TODO add image and timestamp to the display & med_list objects

// TODO sort the med_list in order that user needs to take medication

function HomePage(){

    const [med_list, setMedlist] = useState([
        {id: 0, name: 'med1', isTaken: false},
        {id: 1, name: 'med2', isTaken: false},
        {id: 2, name: 'med3', isTaken: false}
    ]);

    const med_elements = med_list.map((medobj, ind) => {
        const handle_click = (event) => {
            const n = [ ...med_list ];
            n[ind].isTaken = !n[ind].isTaken;
            setMedlist(n);
            console.log(med_list);
        };
        return (
            <button className="link-button font-small" key={medobj.id} onClick={handle_click}>
                {medobj.name}
                <input type="checkbox" value="" className="check-box" checked={medobj.isTaken}></input>
            </button>
        );
    });

    return (
        <div className='background-with-logo-image home-layout'>
            <div className="title">
                <div className="font-very-large">Pill Reminder</div>
                <div className="font-large">Todays pills</div>
            </div>
            <div className="med-list-container flex-container flex-justify-content-space-between">
                {
                    med_elements.map((el) => { return el; })
                }
            </div>
            <div className="button-container flex-container flex-justify-content-end">
                <Link to="/edit" className="link-button font-small">
                    Edit your medication list
                </Link>
            </div>
            <NavBar />
        </div>
    );
}

export { HomePage };