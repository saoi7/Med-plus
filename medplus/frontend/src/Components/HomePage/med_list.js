import React, { useState } from 'react';
import { Link, Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { withFirebase } from '../Firebase';
import NavBar from '../NavBar';
import { withAuthorization } from '../Session';
import * as ROUTES from '../../constants/routes';

function compareDate(a, b) {
    return ((a.getDate() === b.getDay()) && 
            (a.getMonth() === b.getMpnth()) &&
            (a.getFullYear() === b.getFullYear()));
}

function getDateString() {
    let today = new Date();
    // NOTE: .getMonth() starts at 0, so add 1 to get proper month number
    return `${today.getFullYear()}-${today.getMonth()+1}-${today.getDate()}`;
}

function flatten_meds_obj(meds) {
    let med_list = [];
    Object.entries(meds).forEach(entry => {
        let [med_name, val] = entry;
        Object.entries(val).forEach(entry => {
            let [timestamp, obj] = entry;
            let med_list_item = {
                med_name,
                is_taken: obj.is_taken,
                time_to_take: timestamp,
                quantity: obj.quantity,
            };
            med_list.push(med_list_item);
        });
    });
    return med_list;
}

function set_taken_meds_that_arent_in_schedule_obj(meds, taken_meds) {
    Object.entries(taken_meds).forEach((entry) => {
        let [med_name, times_taken_obj] = entry;
        Object.entries(times_taken_obj).forEach((time_obj) => {
            let [time_taken, quantity] = time_obj;
            if(!meds[med_name])
                meds[med_name] = {};
            if(!meds[med_name][time_taken]) {
                // time: { is_taken, quantity } // data structure of meds
                meds[med_name][time_taken] = { is_taken: true, quantity };
            }
        });
    });
    return meds;

}

function set_taken_meds(meds, taken_meds) {
    Object.entries(taken_meds).forEach((entry) => {
        let [med_name, times_taken_obj] = entry;
        Object.entries(times_taken_obj).forEach((time_obj) => {
            let [time_taken, quantity] = time_obj;

            // the if check prevents bugs when you set meds as taken, then change
            // the times for a schedule, then re-load homepage on the same day
            if(meds[med_name] && meds[med_name][time_taken]) {
                meds[med_name][time_taken] = {
                    is_taken: true,
                    quantity: quantity
                };
            }
        });
    });
    return meds;
}

function get_keys_of_todays_meds(meds_obj) {
    let today = new Date(getDateString());  // using getDateString() to create today's date since getDateString() doesn't include the current time in today's date
    let keys_of_todays_meds = Object.keys(meds_obj).filter(key => {
        let start_date = new Date(meds_obj[key].start_date);
        let end_date = new Date(meds_obj[key].end_date);
        return (start_date <= today && today <= end_date);
    });
    return keys_of_todays_meds;
}

function get_todays_meds_obj(meds_obj, filtered_meds_keys) {
    let meds = {};
    filtered_meds_keys.forEach(key => {
        let times_to_take_obj = Object.entries(meds_obj[key].times_to_take);
        times_to_take_obj.forEach((entry, index) => {
            let [time_to_take, quantity] = entry;
            meds[key] = {
                ...meds[key],
                [time_to_take]: {
                    is_taken: false,
                    quantity
                }
            }
        });
    });
    return meds;
}

function sort_meds_list_chronological_order(med_list) {
    med_list.sort((a, b) => {
        if(a.time_to_take < b.time_to_take)
            return -1;
        return 1;
    })
    return med_list;
}

class MedListBase extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            db_ref: null,
            is_done_loading: false,
            firebase_error_flag: false,
            med_list: []
        };
    }

    // TODO
    // show 'no meds today' if everything goes successfully but
    // there are no meds after filtering
    // firebase_error_flag will be false && med_list.length === 0
    populateMedList() {
        this.setState({
            db_ref: this.props.medListDBRef,  // this is necessary for componentDidUpdate() to not enter infinite loop
            is_done_loading: false,
            firebase_error_flag: false,
            med_list: []
        });

        // TODO
        // currently we pull all entries in schedules from firebase
        // then we loop through the entries and filter for entries where today falls between the start_date and end_date
        // see if we can make firebase do the filtering for us (may require restructuring the data in the firebase db)
        // 
        // there are edge cases where you set meds as taken, then modify the schedules object for that med, then check the HomePage again
        if(!this.props.medListDBRef)
            return;
        this.props.medListDBRef.get().then(snapshot => {
            let db_meds_entries = snapshot.val();
            if(!db_meds_entries) {   
                this.setState({ is_done_loading: true });
                return;
            }
            let todays_meds_keys = get_keys_of_todays_meds(db_meds_entries);
            let meds = get_todays_meds_obj(db_meds_entries, todays_meds_keys);
            this.props.firebase.TEST_taken(getDateString()).get().then(snapshot => {
                let taken_meds = snapshot.val();
                if(taken_meds) {
                    meds = set_taken_meds_that_arent_in_schedule_obj(meds, taken_meds);
                    meds = set_taken_meds(meds, taken_meds);
                }
                let med_list = flatten_meds_obj(meds);
                med_list = sort_meds_list_chronological_order(med_list);
                this.setState({ is_done_loading: true, med_list: med_list });
            }).catch(err => {
                this.setState({ is_done_loading: true, firebase_error_flag: true });
            });
        }).catch(err => {
            this.setState({ is_done_loading: true, firebase_error_flag: true });
        });
    }

    componentDidMount() {
        this.populateMedList();
    }

    componentDidUpdate() {
        if(this.state.db_ref !== this.props.medListDBRef)
            this.populateMedList();
    }

    // TODO display prompt if firebase operation failed
    handle_click = ind => event => {
        const med_obj = this.state.med_list[ind];
        const {med_name, time_to_take, quantity} = med_obj;
        const new_is_taken = !med_obj.is_taken;
        const new_val = (new_is_taken) ? quantity : null;  // setting value to null in firebase is equivalent to deleting it
        this.props.firebase.TEST_taken(getDateString()).child(med_name).update({
            [time_to_take]: new_val,
        }).then(arg => {
            const med_list_copy = [ ...this.state.med_list ];
            med_list_copy[ind].is_taken = new_is_taken;
            this.setState({ med_list: med_list_copy });
        });
    }


    // TODO implement better views for loading & error conditions
    render() {
        let result = "loading...";
        if(this.state.is_done_loading && this.state.firebase_error_flag)
            result = "ERROR: failed to load data";
        if(this.state.is_done_loading && !this.state.firebase_error_flag) {
            result = this.state.med_list.map((medobj, ind) => {
                return <MedListItem data={medobj} onClick={this.handle_click(ind)}/>;
            });
        }
        return (
            <div className="med-list-container flex-container flex-justify-content-space-evenly">
                { result }
            </div>
        );
    }
}

function MedListItem(props) {
    const { med_name, time_to_take, is_taken, quantity } = props.data;
    const color_class_str = is_taken ? "background-grey" : "background-blue";
    return (
        <div className="med-list-entry-container font-medium" key={med_name+time_to_take} >
            {time_to_take}
            <button className={"med-list-entry-box font-small " + color_class_str} onClick={props.onClick}>
                <span>{med_name}</span>
                <span>{"Qty: "+quantity}</span>
                <input type="checkbox" value="" className="check-box" checked={is_taken} />
            </button>
        </div>
    );
}

const MedList = withFirebase(MedListBase);

export default MedList;