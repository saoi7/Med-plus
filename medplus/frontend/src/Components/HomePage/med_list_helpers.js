function compareDate(a, b) {
    return ((a.getDate() === b.getDay()) && 
            (a.getMonth() === b.getMpnth()) &&
            (a.getFullYear() === b.getFullYear()));
}

function deep_copy(obj_to_be_copied) {
    return JSON.parse(JSON.stringify(obj_to_be_copied));
}

function getDateString() {
    let today = new Date();
    // NOTE: .getMonth() starts at 0, so add 1 to get proper month number
    return `${today.getFullYear()}-${today.getMonth()+1}-${today.getDate()}`;
}

function flatten_meds_obj(meds) {
    let med_list = [];
    Object.entries(meds).forEach(entry => {
        let [med_name, val] = entry;
        Object.entries(val).forEach(entry => {
            let [timestamp, obj] = entry;
            let med_list_item = {
                med_name,
                is_taken: obj.is_taken,
                time_to_take: timestamp,
                quantity: obj.quantity,
            };
            med_list.push(med_list_item);
        });
    });
    return med_list;
}

function set_taken_meds_that_arent_in_schedule_obj(meds, taken_meds) {
    let meds_copy = deep_copy(meds);
    Object.entries(taken_meds).forEach((entry) => {
        let [med_name, times_taken_obj] = entry;
        Object.entries(times_taken_obj).forEach((time_obj) => {
            let [time_taken, quantity] = time_obj;
            if(!meds_copy[med_name])
                meds_copy[med_name] = {};
            if(!meds_copy[med_name][time_taken]) {
                // time: { is_taken, quantity } // data structure of meds
                meds_copy[med_name][time_taken] = { is_taken: true, quantity };
            }
        });
    });
    return meds_copy;
}

function set_taken_meds(meds, taken_meds) {
    let meds_copy = deep_copy(meds);
    Object.entries(taken_meds).forEach((entry) => {
        let [med_name, times_taken_obj] = entry;
        Object.entries(times_taken_obj).forEach((time_obj) => {
            let [time_taken, quantity] = time_obj;

            // the if check prevents bugs when you set meds as taken, then change
            // the times for a schedule, then re-load homepage on the same day
            if(meds_copy[med_name] && meds_copy[med_name][time_taken]) {
                meds_copy[med_name][time_taken] = {
                    is_taken: true,
                    quantity: quantity
                };
            }
        });
    });
    return meds_copy;
}

function get_keys_of_todays_meds(meds_obj) {
    let today = new Date(getDateString());  // using getDateString() to create today's date since getDateString() doesn't include the current time in today's date
    let keys_of_todays_meds = Object.keys(meds_obj).filter(key => {
        let start_date = new Date(meds_obj[key].start_date);
        let end_date = new Date(meds_obj[key].end_date);
        return (start_date <= today && today <= end_date);
    });
    return keys_of_todays_meds;
}

function get_todays_meds_obj(meds_obj, filtered_meds_keys) {
    let meds = {};
    filtered_meds_keys.forEach(key => {
        let times_to_take_obj = Object.entries(meds_obj[key].times_to_take);
        times_to_take_obj.forEach((entry, index) => {
            let [time_to_take, quantity] = entry;
            meds[key] = {
                ...meds[key],
                [time_to_take]: {
                    is_taken: false,
                    quantity
                }
            }
        });
    });
    return meds;
}

function sort_meds_list_chronological_order(med_list) {
    let med_list_copy = deep_copy(med_list);
    med_list_copy.sort((a, b) => {
        if(a.time_to_take < b.time_to_take)
            return -1;
        return 1;
    })
    return med_list_copy;
}

export { 
    getDateString,
    flatten_meds_obj,
    set_taken_meds_that_arent_in_schedule_obj,
    set_taken_meds,
    get_keys_of_todays_meds,
    get_todays_meds_obj,
    sort_meds_list_chronological_order
};
