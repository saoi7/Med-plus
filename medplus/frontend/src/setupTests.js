// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
