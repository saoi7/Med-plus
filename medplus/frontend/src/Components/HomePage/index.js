import {
    getDateString,
    flatten_meds_obj,
    set_taken_meds_that_arent_in_schedule_obj,
    set_taken_meds,
    get_keys_of_todays_meds, 
    get_todays_meds_obj,
    sort_meds_list_chronological_order
} from './med_list_helpers';
import { MedListBase, MedListItem } from './med_list';
import MedList from './med_list';
import HomePage from './homepage';
import { HomePageBase, MedListSelection } from './homepage';

export default HomePage;
export { HomePageBase, MedListSelection };
export { getDateString, flatten_meds_obj, set_taken_meds_that_arent_in_schedule_obj, set_taken_meds, get_keys_of_todays_meds, get_todays_meds_obj, sort_meds_list_chronological_order };
export { MedListBase, MedListItem };
export { MedList };
