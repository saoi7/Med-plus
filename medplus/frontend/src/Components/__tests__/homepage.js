import {
    getDateString,
    flatten_meds_obj,
    set_taken_meds_that_arent_in_schedule_obj,
    set_taken_meds,
    get_keys_of_todays_meds,
    get_todays_meds_obj,
    sort_meds_list_chronological_order
} from '../HomePage';
import { MOCK_DB_MED_ENTRIES } from './mock_db_med_entries';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import { getByTestId, fireEvent } from '@testing-library/react';
import { MedListBase, MedListItem, HomePageBase, MedListSelection } from '../HomePage';
import { MemoryRouter } from 'react-router-dom';

// getDateString() will return a different string depending on the current date.
// mocking getDateString() function so that it will always return the same date
// so that tests will work the same no matter when you run them
// see: https://jestjs.io/docs/mock-functions#mocking-partials
jest.mock('../HomePage/med_list_helpers', () => {
    const ORIGINAL_MODULE = jest.requireActual('../HomePage/med_list_helpers');
    return {
        ...ORIGINAL_MODULE,
        getDateString: () => '2021-12-05'
    };
});

test("unit tests of get_keys_of_todays_meds", () => {
    const EXPECTED_MED_KEYS = ['abc', 'ghi'];
    const MED_KEYS = get_keys_of_todays_meds(MOCK_DB_MED_ENTRIES);
    expect(MED_KEYS.length).toBe(2);
    expect(MED_KEYS).toEqual(expect.arrayContaining(EXPECTED_MED_KEYS));
});

test("unit tests of get_todays_meds_obj", () => {
    const MOCK_MED_KEYS = ['abc', 'ghi'];
    const MEDS = get_todays_meds_obj(MOCK_DB_MED_ENTRIES, MOCK_MED_KEYS);
    expect(Object.keys(MEDS).length).toBe(2);

    expect(MEDS.abc['04:00'].is_taken).toBe(false);
    expect(MEDS.abc['04:00'].quantity).toBe('1');

    expect(MEDS.abc['05:00'].is_taken).toBe(false);
    expect(MEDS.abc['05:00'].quantity).toBe('999');

    expect(MEDS.ghi['12:34'].is_taken).toBe(false);
    expect(MEDS.ghi['12:34'].quantity).toBe('1');
});

test("unit tests of set_taken_meds_that_arent_in_schedule_obj", () => {
    const MOCK_MEDS = {
        abc: {
            '04:00': { is_taken: false, quantity: '1' },
            '05:00': { is_taken: false, quantity: '999' },
        },
        ghi: {
            '12:34': { is_taken: false, quantity: '1' }
        },
    };
    const MOCK_TAKEN_MEDS = {
        abc: {
            "05:00": "999"
        },
        ghi: {
            "12:34": "1"
        },
        zzz: {
            "01:00": "3"
        }
    };
    const MEDS = set_taken_meds_that_arent_in_schedule_obj(MOCK_MEDS, MOCK_TAKEN_MEDS);
    expect(Object.keys(MEDS).length).toBe(3);

    expect(MEDS.abc['04:00'].is_taken).toBe(false);
    expect(MEDS.abc['04:00'].quantity).toBe('1');

    expect(MEDS.abc['05:00'].is_taken).toBe(false);
    expect(MEDS.abc['05:00'].quantity).toBe('999');

    expect(MEDS.ghi['12:34'].is_taken).toBe(false);
    expect(MEDS.ghi['12:34'].quantity).toBe('1');

    expect(MEDS.zzz['01:00'].is_taken).toBe(true);
    expect(MEDS.zzz['01:00'].quantity).toBe('3');
});

test("unit tests of set_taken_meds", () => {
    const MOCK_MEDS = {
        abc: {
            '04:00': { is_taken: false, quantity: '1' },
            '05:00': { is_taken: false, quantity: '999' },
        },
        ghi: {
            '12:34': { is_taken: false, quantity: '1' }
        },
        zzz: {
            '01:00': { is_taken: true, quantity: '3' }
        }
    };
    const MOCK_TAKEN_MEDS = {
        abc: {
            "05:00": "999"
        },
        ghi: {
            "12:34": "999" // note this is different from MOCK_MEDS to represent that user has changed quantity
        },
        zzz: {
            "01:00": "3"
        }
    };
    const MEDS = set_taken_meds(MOCK_MEDS, MOCK_TAKEN_MEDS);
    expect(Object.keys(MEDS).length).toBe(3);

    expect(MEDS.abc['04:00'].is_taken).toBe(false);
    expect(MEDS.abc['04:00'].quantity).toBe('1');

    expect(MEDS.abc['05:00'].is_taken).toBe(true);
    expect(MEDS.abc['05:00'].quantity).toBe('999');

    expect(MEDS.ghi['12:34'].is_taken).toBe(true);
    expect(MEDS.ghi['12:34'].quantity).toBe('999');

    expect(MEDS.zzz['01:00'].is_taken).toBe(true);
    expect(MEDS.zzz['01:00'].quantity).toBe('3');
});

test("unit tests of flatten_meds_obj", () => {
    const MOCK_MEDS = {
        abc: {
            '04:00': { is_taken: true, quantity: '2' },
        },
        ghi: {
            '12:34': { is_taken: false, quantity: '1' }
        },
        zzz: {
            '01:00': { is_taken: true, quantity: '3' }
        }
    };
    const MEDS = flatten_meds_obj(MOCK_MEDS);
    expect(MEDS.length).toBe(3);

    expect(MEDS[0].med_name).toBe('abc');
    expect(MEDS[0].is_taken).toBe(true);
    expect(MEDS[0].time_to_take).toBe('04:00');
    expect(MEDS[0].quantity).toBe('2');

    expect(MEDS[1].med_name).toBe('ghi');
    expect(MEDS[1].is_taken).toBe(false);
    expect(MEDS[1].time_to_take).toBe('12:34');
    expect(MEDS[1].quantity).toBe('1');

    expect(MEDS[2].med_name).toBe('zzz');
    expect(MEDS[2].is_taken).toBe(true);
    expect(MEDS[2].time_to_take).toBe('01:00');
    expect(MEDS[2].quantity).toBe('3');
});

test("unit tests of sort_meds_list_chronological_order", () => {
    const MOCK_FLATTENED_MEDS_OBJ = [
        {
            med_name: 'abc',
            is_taken: true,
            time_to_take: '14:11',
            quantity: '2'
        },
        {
            med_name: 'ghi',
            is_taken: false,
            time_to_take: '12:34',
            quantity: '1'
        },
        {
            med_name: 'zzz',
            is_taken: true,
            time_to_take: '01:00',
            quantity: '3'
        }
    ];

    const MEDS = sort_meds_list_chronological_order(MOCK_FLATTENED_MEDS_OBJ);
    expect(MEDS.length).toBe(3);

    expect(MEDS[0].med_name).toBe('zzz');
    expect(MEDS[0].is_taken).toBe(true);
    expect(MEDS[0].time_to_take).toBe('01:00');
    expect(MEDS[0].quantity).toBe('3');

    expect(MEDS[1].med_name).toBe('ghi');
    expect(MEDS[1].is_taken).toBe(false);
    expect(MEDS[1].time_to_take).toBe('12:34');
    expect(MEDS[1].quantity).toBe('1');

    expect(MEDS[2].med_name).toBe('abc');
    expect(MEDS[2].is_taken).toBe(true);
    expect(MEDS[2].time_to_take).toBe('14:11');
    expect(MEDS[2].quantity).toBe('2');
});

const MOCK_FIREBASE_DB_REF_PROP = {
    get: () => { return Promise.resolve({ val: MOCK_DB_MED_ENTRIES }); }
};
const MOCK_TAKEN_MEDS = {
    abc: {
        "05:00": "999"
    },
    ghi: {
        "12:34": "999" // note this is different from MOCK_MEDS to represent that user has changed quantity
    },
    zzz: {
        "01:00": "3"
    }
};
const MOCK_FIREBASE_PROP = {
    currentUserUID: () => 123,
    currentUser: () => {
        return {
            get: () => {
                return Promise.resolve({ username: 'abc' });
            }
        };
    },
    getShareeOfRef: () => {
        return {
            get: () => {
                // returning empty object since the feature that uses this obj ended up not getting fully implement
                return Promise.resolve({});
            }
        }
    },
    TEST_schedules: () => MOCK_FIREBASE_DB_REF_PROP,
    TEST_taken: (date_str) => {
        return {
            get: () => {
                return Promise.resolve({ val: MOCK_TAKEN_MEDS });
            },
            child: (med_name) => {
                return {
                    update: () => {
                        return Promise.resolve();
                    }
                }
            }
        }
    }
};

test("snapshot test of MedList to prevent unintentional changes", () => {
    const COMPONENT= renderer.create(
        <MedListBase firebase={MOCK_FIREBASE_PROP} medListDBRef={MOCK_FIREBASE_DB_REF_PROP} />
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});

test("snapshot test of MedListItem to prevent unintentional changes", () => {
    const PROP_1 = {
        med_name: 'abc',
        time_to_take: '01:00',
        is_taken: true,
        quantity: '123'
    }

    const COMPONENT_1 = renderer.create(
        <MedListItem data={PROP_1} />
    );
    let tree = COMPONENT_1.toJSON();
    expect(tree).toMatchSnapshot();

    const PROP_2 = {
        med_name: 'e9rtjqn32knto34ijy',
        time_to_take: '12:34',
        is_taken: false,
        quantity: '999'
    }

    const COMPONENT_2 = renderer.create(
        <MedListItem data={PROP_2} />
    );
    tree = COMPONENT_2.toJSON();
    expect(tree).toMatchSnapshot();
});

test("snapshot test of HomePage to prevent unintentional changes", () => {
    const COMPONENT = renderer.create(
        <MemoryRouter>
            <HomePageBase firebase={MOCK_FIREBASE_PROP} />
        </MemoryRouter>
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});

test("snapshot test of MedListSelection to prevent unintentional changes", () => {
    const COMPONENT = renderer.create(
        <MedListSelection medListObjs={{ 'abc': {ref: 'zzz', uid: 123 } }} />
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});
