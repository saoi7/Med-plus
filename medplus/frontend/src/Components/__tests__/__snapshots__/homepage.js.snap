// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of HomePage to prevent unintentional changes 1`] = `
<div
  className="background-with-logo-image home-layout"
>
  <div
    className="med-list-selection"
  >
    
  </div>
  <div
    className="title"
  >
    <div
      className="font-very-large"
    >
      Med Reminder
    </div>
    <div
      className="font-large"
    >
      Todays Meds:
    </div>
  </div>
  <div
    className="med-list-container flex-container flex-justify-content-space-evenly"
  >
    loading...
  </div>
  <div
    className="button-container flex-container flex-justify-content-end"
  >
    <a
      className="link-button font-small"
      href="/manage"
      onClick={[Function]}
    >
      Edit your medication list
    </a>
  </div>
  <nav
    className="nav-bar"
  >
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      Reminders
    </a>
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
        />
      </svg>
    </a>
    <a
      className="font-medium"
      href="/account"
      onClick={[Function]}
    >
      Profile
    </a>
  </nav>
</div>
`;

exports[`snapshot test of MedList to prevent unintentional changes 1`] = `
<div
  className="med-list-container flex-container flex-justify-content-space-evenly"
>
  loading...
</div>
`;

exports[`snapshot test of MedListItem to prevent unintentional changes 1`] = `
<div
  className="med-list-entry-container font-medium"
>
  01:00
  <button
    className="med-list-entry-box font-small background-grey"
  >
    <span>
      abc
    </span>
    <span>
      Qty: 123
    </span>
    <input
      checked={true}
      className="check-box"
      type="checkbox"
      value=""
    />
  </button>
</div>
`;

exports[`snapshot test of MedListItem to prevent unintentional changes 2`] = `
<div
  className="med-list-entry-container font-medium"
>
  12:34
  <button
    className="med-list-entry-box font-small background-blue"
  >
    <span>
      e9rtjqn32knto34ijy
    </span>
    <span>
      Qty: 999
    </span>
    <input
      checked={false}
      className="check-box"
      type="checkbox"
      value=""
    />
  </button>
</div>
`;

exports[`snapshot test of MedListSelection to prevent unintentional changes 1`] = `
<div
  className="med-list-selection"
>
  
</div>
`;
